airbyte-integrations/connectors/destination-snowflake/Dockerfile

```diff
@@ -11,6 +11,9 @@ WORKDIR /airbyte
 
 ENV APPLICATION destination-snowflake
 
+# Needed for JDK17 (in turn, needed on M1 macs) - see https://github.com/snowflakedb/snowflake-jdbc/issues/589#issuecomment-983944767
+ENV DESTINATION_SNOWFLAKE_OPTS "--add-opens java.base/java.nio=ALL-UNNAMED"
```

Comment:
@edgao looks like issue described [here](https://github.com/snowflakedb/snowflake-jdbc/issues/589#issuecomment-983944767) still persists in master branch

Reply:
yeah, I'm not sure what happened - snowflake was definitely passing for me locally, but after the java 17 upgrade it's failing again. Haven't been able to find a fix, either.

Reply:
@edgao @VitaliiMaltsev Were you guys able to find the fix?

Reply:
https://github.com/airbytehq/airbyte/pull/8781/files#diff-0fd1ff4a11af6edb89860081578c13598aa6cdb289c3e6ceae2a9d44eb708b80R45-R46 was the fix

(also https://github.com/airbytehq/airbyte/pull/8854/files#diff-a3dd6be9c3819c1fcfd593b0f11c29663912e686e0dd74cae14610ac9c282480R50, but technically we only needed to one or the other)

Reply:
@edgao is there any side effect using JVM command `jvmArgs = ["--add-opens=java.base/java.nio=ALL-UNNAMED"]`  ? 
Any potential security issues?

Reply:
my understanding is that this isn't a security check, it's more so that the Java devs can modify some Java internal implementation in the future. I.e. they want to discourage reliance on java's internal APIs, so access is now disabled by default.

Reply:
@edgao thanks for the response. However, based on the [JEPS-261](https://openjdk.java.net/jeps/261):

> The --add-exports and --add-opens options must be used with great care. You can use them to gain access to an internal API of a library module, or even of the JDK itself, but you do so at your own risk: If that internal API is changed or removed then your library or application will fail.

I'm not that familiar with how gaining more internal access to java.nio could be concerning. Therefore, asking if you have insight into it.

Reply:
The risk occurs when upgrading Java versions - there's no guarantee that the underlying implementation will remain the same. So in theory, the Java devs could release a new version of Java, which modifies those internal APIs, and therefore break the snowflake JDBC driver. For Airbyte, this risk is mitigated by the source/destination acceptance tests, which actually build the Docker image and check that it can interact with a real Snowflake instance, i.e. this would be caught before release. But we would be unable to upgrade to that new Java version  until an updated snowflake-jdbc is released.

I'm not actually sure which issue is being used to track progress on removing snowflake-jdbc's dependency on those internal APIs, but would recommend asking in https://github.com/snowflakedb/snowflake-jdbc if you're interested in the details / how they're thinking about the problem. Some potentially related issues: https://github.com/snowflakedb/snowflake-jdbc/issues/484, https://github.com/snowflakedb/snowflake-jdbc/issues/589, https://github.com/snowflakedb/snowflake-jdbc/issues/533